### PR TITLE
Reduce overhead

### PR DESCRIPTION
**Pull request by rowleya**
Add checksums to regions

### CODE REVIEW

File: data_specification/constants.py [hunk 1]
```diff
@@ -50,7 +50,8 @@
 #: Size of header of data spec pointer table produced by DSE, in bytes.
 APP_PTR_TABLE_HEADER_BYTE_SIZE = 8
 #: Size of data spec pointer table produced by DSE, in bytes.
-APP_PTR_TABLE_BYTE_SIZE = APP_PTR_TABLE_HEADER_BYTE_SIZE + MAX_MEM_REGIONS * 4
+APP_PTR_TABLE_BYTE_SIZE = (
+    APP_PTR_TABLE_HEADER_BYTE_SIZE + MAX_MEM_REGIONS * 4 * 3)
```
Comment (Christian-B) [inline data_specification/constants.py:54]: Magic numbers.
The 4 I imagine is bytes to word which is ok
But why the 3?

Comment (Christian-B) [reply data_specification/constants.py:54]: Similar in Constants.java

Comment (Christian-B) [reply data_specification/constants.py:54]: I see it is correct as it covers  "pointer", "n_words" and "checksum" 
But I should not have had to look!

**Review by Christian-B — CHANGES_REQUESTED** [review]

File: data_specification/constants.py (hunk 1 above)
Comment (rowleya) [reply data_specification/constants.py]: Fixed

**Review by Christian-B — APPROVED** [review]

**Pull request merged**